src/hotspot/share/utilities/vmError.cpp

```diff
@@ -547,19 +547,26 @@ void VMError::report(outputStream* st, bool _verbose) {
   // test secondary error handling. Test it twice, to test that resetting
   // error handler after a secondary crash works.
   STEP("test secondary crash 1")
-    if (_verbose && TestCrashInErrorHandler != 0) {
+    if (_verbose && TestCrashInErrorHandler == 14) {
```

Comment:
Where does the 14 come from ??

Reply:
Maybe "// 14 - SIGSEGV" in vmError.cpp?

Reply:
See runtime/ErrorHandling/SecondaryErrorTest.java, it is fed into VMError::controlled_crash().

That test crashes the VM first with one signal, then inside the error handler with a second, different, signal, to test that the second error signal was correctly unblocked or left unblocked. Otherwise, the VM would just disappear.

Reply:
Yes, the 14 comes with the test.  Nothing else queries the value of TestCrashInErrorHandler or passes anything but 14 or now in my test, 2.

Reply:
We could change the 14 to the first test to some enum test_error_handler { secondary_crash = 1, resource_mark_crash = 2 }; and change the test.  I just left the 14 since the test can't see the enum values.

Reply:
I gave them local #defines so they'd have good names.